eng/performance/benchmark_jobs.yml

```
@@ -33,7 +34,7 @@ jobs:
             value: dotnet-performance
           # for public runs we want to run the benchmarks exactly once, no warmup, no pilot, no overhead
           - name: BenchmarkDotNetArguments
-            value: '--allCategories $(_Category) --iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart --stopOnFirstError true'
+            value: '--anyCategories ${{ parameters.runCategories }} --iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart --stopOnFirstError true'
```

I think that it can be simplified and we can filter by providing `--filter *` instead of `--anyCategories`

So far we have 3 projects with BDN benchmarks:

- Roslyn with only Roslyn benchmarks
- ML.NET with only ML.NET benchmarks
- MicroBenchmarks with CoreFX, CoreCLR and ThirdParty benchmarks. The benchmarks from `ThirdParty` category are 40-60 benchmarks for the most popular 3rd party serializers that we have excluded before to save some time when running the benchmarks as part of the CI. But now when we have the possibility to parition them and send to multiple machines and imho we don't need to do this anymore and we can run all of them.

```suggestion
            value: "--filter '*' --iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart --stopOnFirstError true"
```

The asterisk needs to be wrapped with single quote characters on Unix because otherwise the OS replaces it with the list of files in current folder.

---

We tried that. There are many benchmarks in the microbenchmark project that are not triggered when you specify coreclr or corefx that are triggered by filter *. The serializer benchmarks are broken, which causes everything to break. They need to be fixed before we can include them. Is there another way to disable benchmarks while using filter *?